File: remove-stones-to-minimize-the-total/remove-stones-to-minimize-the-total.cs

public class Solution {
    public int MinStoneSum(int[] piles, int k) {
        
        //PriorityQueue<int,int> maxHeap=new PriorityQueue<int,int>(Comparer<int>.Create((x,y)=>y.CompareTo(x)));
        PriorityQueue<int,int> maxHeap=new PriorityQueue<int,int>();

        int sum=0;
        
        for(int i=0;i<piles.Length;i++)
        {
            maxHeap.Enqueue(piles[i],-1*piles[i]);
            sum=sum+piles[i];
        }
        
        int remaining=0;
        while(k>0)
        {
            int HeaviestStone=maxHeap.Dequeue();
             sum=sum-HeaviestStone/2;
            remaining=HeaviestStone-HeaviestStone/2;
            maxHeap.Enqueue(remaining,-1*remaining);
            k--;
        }
        
        return sum;
        
        
    }
}

File: remove-stones-to-minimize-the-total/remove-stones-to-minimize-the-total.java
class Solution {
    public int minStoneSum(int[] piles, int k) {
        
        PriorityQueue<Integer> queue=new PriorityQueue<>(Comparator.reverseOrder());
        
        for(int i=0;i<piles.length;i++)
        {
            queue.add(piles[i]);
        }
        
        int ans=0;
        while(k>0)
        {
            int num=queue.remove();
            
            queue.add(num-num/2);
            
            k--;
        }
        
        while(queue.size()>0)
        {
            ans=ans+queue.remove();
        }
        
        return ans;
        
    }
}